diff -ur libcgroup-0.37/samples/cgconfig.conf libcgroup-0.37-conf/samples/cgconfig.conf
--- libcgroup-0.37/samples/cgconfig.conf	2010-10-20 15:59:13.000000000 +0200
+++ libcgroup-0.37-conf/samples/cgconfig.conf	2011-02-10 18:05:17.560363010 +0100
@@ -10,39 +10,19 @@
 #  WITHOUT ANY WARRANTY; without even the implied warranty of
 #  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 #
-#group daemons/www {
-#	perm {
-#		task {
-#			uid = root;
-#			gid = webmaster;
-#		}
-#		admin {
-#			uid = root;
-#			gid = root;
-#		}
-#	}
-#	cpu {
-#		cpu.shares = 1000;
-#	}
-#}
+# See man cgconfig.conf for further details.
 #
-#group daemons/ftp {
-#	perm {
-#		task {
-#			uid = root;
-#			gid = ftpmaster;
-#		}
-#		admin {
-#			uid = root;
-#			gid = root;
-#		}
-#	}
-#	cpu {
-#		cpu.shares = 500;
-#	}
-#}
-#
-#mount {
-#	cpu = /mnt/cgroups/cpu;
-#	cpuacct = /mnt/cgroups/cpuacct;
-#}
+# By default, mount all controllers to /sys/fs/cgroup/<controller>
+#
+#mount {
+#	blkio   = /sys/fs/cgroup/blkio;
+#	cpu     = /sys/fs/cgroup/cpu;
+#	cpuacct = /sys/fs/cgroup/cpuacct;
+#	cpuset  = /sys/fs/cgroup/cpuset;
+##	debug   = /sys/fs/cgroup/debug;
+#	devices = /sys/fs/cgroup/devices;
+#	freezer = /sys/fs/cgroup/freezer;
+#	memory  = /sys/fs/cgroup/memory;
+#	net_cls = /sys/fs/cgroup/net_cls;
+#}
+
diff -ur libcgroup-0.37/samples/cgconfig.sysconfig libcgroup-0.37-conf/samples/cgconfig.sysconfig
--- libcgroup-0.37/samples/cgconfig.sysconfig	2010-10-20 15:59:13.000000000 +0200
+++ libcgroup-0.37-conf/samples/cgconfig.sysconfig	2011-02-10 18:06:29.233508500 +0100
@@ -5,8 +5,5 @@
 # controller to limit cpu.shares of this default group and allowing some more
 # important group take most of the CPU.
 #
-# By default, create these groups:
-CREATE_DEFAULT=yes
-
-# Uncomment following line to disable creation of the default group on startup:
-# CREATE_DEFAULT=no
+# Don't create default group on startup
+CREATE_DEFAULT=no
